### ISSUE TITLE
Support custom base for `at.log` function?

### ISSUE THREAD

**Issue by zoj613** [aesara-devs/aesara#793]
## Description of your problem or feature request

im wondering if it worth supporting a custom base for `at.log` just like how `math.log` is able to. We can then represent ratio of two logs succinctly  using the relation: `log_c(a) / log_c(b) = log_b(a)`
This was inspired by a recent comment I made here: https://github.com/pymc-devs/pymc/issues/5078#issuecomment-1022019224
Thoughts? CC @brandonwillard @ricardoV94 @kc611 @rlouf

**Comment by rlouf** [comment]
imho in doubt we should always _do like numpy_ which doesn't have the best possible API, but has a good enough API everyone is used to. See the [log function signature](https://numpy.org/doc/stable/reference/generated/numpy.log.html).

**Comment by ricardoV94** [comment]
I don't see the problem with adding an additional parameter for the base. We can make `log` be a function that returns the relevant expression or the pre-builtins `log`, `log2` `log10`.

**Comment by rlouf** [comment]
I would add a `logb` function or something similar. I would find it semantically weird to add a `base` argument to `log`, it changes the function's meaning imo.

**Comment by ricardoV94** [comment]
Why? `at.log(x, base=np.e)` seems pretty logical.

**Comment by rlouf** [comment]
Because in my head `log` = logarithm base e. Would you do `np.exp(x, base=2)` to represent `2^x`?

**Comment by zoj613** [comment]
> Because in my head `log` = logarithm base e. Would you do `np.exp(x, base=2)` to represent `2^x`?

I believe the addition of an optional parameter `base` should not change the default meaning. It only changes when the user specifically _wants_ to use a different base. In that case the intent would be communicated easily by the addition of the parameter in the call.

**Comment by ricardoV94** [comment]
> Because in my head `log` = logarithm base e. Would you do `np.exp(x, base=2)` to represent `2^x`?

I think I would actually... if there was no Python operator support to express that already

**Comment by kc611** [comment]
IMO the better way here, Implementation-wise, would be to simply have an `logb` helper function which would convert the base into another log in denominator. Because in the actual `Op.perform` we'd anyways would have to do something like `np.log(array) / np.log(base)` if we are to calculate the log of Numpy arrays. Might as well have them bring two different operations (that could possibly get optimized out which may not be the case when they are represented as a single `Op`)

**Comment by rlouf** [comment]
> I think I would actually... if there was no Python operator support to express that already

Would you or would you implement `power`?

**Comment by ricardoV94** [comment]
@kc611 I don't think we should create new `Op`s necessarily. The idea would be to return the division behind the scenes for special cases, or the basic `log`, `log2`, `log10` for the pre-built cases

**Comment by kc611** [comment]
> @kc611 I don't think we should create new Ops necessarily. The idea would be to return the division behind the scenes for special cases, or the basic log, log2, log10 for the pre-built cases

Ah ok, I think we're taking about the same thing.

**Comment by rlouf** [comment]
> It only changes when the user specifically _wants_ to use a different base. In that case the intent would be communicated clearly by the addition of the parameter in the call.

My rule of thumb is that `kwargs` should never change the established meaning of a function. I'm sure there are some good counter examples out there, but it has served me well so far.
